catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/geofeature/Feature.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.geofeature;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Base class for results returned from the FeatureService.
+ */
+public abstract class Feature {
```

Comment:
So, it may be more trouble than it is worth, not sure.  The challenge is always the circle.  Ultimately, you are creating these so some service can serve them out right?  And you are serving them out by doing a toJson of these objects.  So the real goal is to serialize out geometries to the system.

We tend to work with three standard for these geometries: geoJson (json), gml (xml), and wkt (human readable).

gml has the best representation of a circle with a few different options, the most obvious being CircleByCenterPoint.  geojson doesn't have a circle but it has a Feature which includes a geometry (which could be a point) and properties (which could include a radius).  There is an old proposal to add circles and ellipses, but it seems to be going no where.  WKT doesn't really have anything, and often uses polygon approximations.

JTS seems to follow closely to what WKT can support, and I don't really see much circle support there.  Again, it relies on polygon approximations.

So, the answer might well be "we do need to do something like this"...